gslab_scons/tests/test_size_warning.py

```diff
@@ -0,0 +1,312 @@
+            sizes = sw.create_size_dictionary(['nonexistent_directory'])
+        # The path argument must be a string
+        with self.assertRaises(TypeError), nostderrout():
+            sizes = sw.create_size_dictionary([10])
+
+
+#== Side effects for testing list_ignored_files() ===
+# Define the mock file structure for testing list_ignored_files()
+struct = {'.': ['untracked.txt', 'make.log', 'make.py'],
+         'raw': ['large_file.txt', 'small_file.txt'], 
+         'release': ['output.txt', '.DS_Store'],
+         'release/subdir': ['ignored.txt']}
+
+def check_ignored_side_effect(ignored = 'standard'):
+
+    def effect(*args, **kwargs):
+        '''Mock subprcess.check_output() for testing list_ignored_files()'''
+    
+        # Define mock messages from "git status --ignored"
+        # i) Some files are ignored
+        standard = \
+            ['On branch testing\n',
+             'Your branch is up-to-date with \'origin/testing\'.\n',
+             'Changes not staged for commit:\n',
+             '  (use "git add/rm <file>..." to update what will be committed)\n',
+             '  (use "git checkout -- <file>..." to discard changes in working directory)\n',
+             '\n',
+             '\tmodified:   make.log\n',
+             '\n',
+             'Untracked files:\n',
+             '  (use "git add <file>..." to include in what will be committed)\n',
+             '\n',
+             '\tuntracked.txt',
+             '\n',
+             'Ignored files:\n',
+             '  (use "git add -f <file>..." to include in what will be committed)\n',
+             '\n',
+             '\traw/large_file.txt\n',
+             '\trelease/.DS_Store\n',
+             '\trelease/subdir/'
+             '\n',
+             '\n',
+             'It took 2.44 seconds to enumerate untracked files. \'status -uno\'\n',
+             'may speed it up, but you have to be careful not to forget to add\n',
+             'new files yourself (see \'git help status\').\n',
+             'no changes added to commit (use "git add" and/or "git commit -a")\n']
+        
+        # ii) No files are ignored
+        none_ignored = \
+            ['On branch issue59-size_warnings\n',
+             'Your branch is up-to-date with \'origin/issue59-size_warnings\'.\n',
+             'nothing to commit, working tree clean\n']
+
+        if ignored == 'none':
+            message = ''.join(none_ignored)
+        else:
+            message = ''.join(standard)
+
+        command = args[0]
+        if 'shell' in kwargs.keys():
+            if kwargs['shell'] and (command == 'git status --ignored'):
+                return message
+    
+        return None
+
+    return effect
+
+
+def walk_ignored_side_effect(*args, **kwargs):
+    '''Mock os.walk() for testing list_ignored_files()'''
+    path = args[0]
+    path = os.path.relpath(path)
+
+    if path not in struct.keys():
+        raise StopIteration
+
+    if path == 'release':
```

Comment:
You're right. I never caught this because it doesn't matter for the mock in question (we never walk `.`) but I'll correct it now.